src/Functions/URL/netloc.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "FunctionsURL.h"
+#include <common/find_symbols.h>
+
+
+namespace DB
+{
+
+struct ExtractNetloc
+{
+    /// We use the same as domain function
+    static size_t getReserveLengthForElement() { return 15; }
+
+    static void execute(Pos data, size_t size, Pos & res_data, size_t & res_size)
+    {
+        res_data = data;
+        res_size = size;
+
+        Pos pos = data;
+        Pos end = pos + size;
+
+        pos = find_first_symbols<'/'>(pos, end);
+        if (end == pos)
+            return;
+
+        /// Strings are zero-terminated.
+        bool has_subsequent_slash = pos[1] == '/';
+        if (!has_subsequent_slash)
+            return;
+        res_data = pos + 2;
+        res_size = end - res_data;
+
+        /// Search for next slash or question mark
+        /// Note than currently the netloc function doesn't support
+        /// if we have a question mark as an username or password.
+        /// This choice has been made for not hurting performance with still taking into
+        /// acount URL without a slash in the end of a query, with a query string.
+        pos = find_first_symbols<'/', '?'>(pos + 2, end);
+        if (end == pos)
+            return;
+        res_size = pos - res_data;
```

Comment:
AFAICS there is no checks against restricted symbols, like in domain.h:

https://github.com/ClickHouse/ClickHouse/blob/58786f992988a16a8ea787fe624c791529343702/src/Functions/URL/domain.h#L51

Also maybe it will be better to add function to extract `userinfo` instead and `netloc()` will be `concat(userinfo(url), domain(url))` (yep, will do extra job) ?

Reply:
Well it's unlikely that someone will use `FixedString` for keeping URLs and moreover that `FixedString` will end with slash. I believe it's just more correct to check boundaries.